// piral-debug-utils/esm/debug.js
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
export function installPiralDebug(options) {
    const { context } = options, pilets = __rest(options, ["context"]);
    // the DEBUG_PIRAL env should contain the Piral CLI compatibility version
    window['dbg:piral'] = {
        debug: 'v0',
        instance: {
            name: process.env.BUILD_PCKG_NAME,
            version: process.env.BUILD_PCKG_VERSION,
            dependencies: process.env.SHARED_DEPENDENCIES,
            context,
        },
        build: {
            date: process.env.BUILD_TIME_FULL,
            cli: process.env.PIRAL_CLI_VERSION,
            compat: process.env.DEBUG_PIRAL,
        },
        pilets,
    };
}
